packages/modals/src/styled/StyledDrawerModalHeader.ts

```diff
@@ -16,6 +17,13 @@ export const StyledDrawerModalHeader = styled(StyledHeader).attrs({
   'data-garden-version': PACKAGE_VERSION
 })`
   padding: ${props => props.theme.space.base * 5}px;
+  ${props =>
+    props.isCloseButtonPresent &&
+    `padding-${props.theme.rtl ? 'left' : 'right'}: ${
+      // the padding added to the Header is based on the close button size and spacing,
+      // with additional padding (+ 2) between the Header content and Close button
```

Comment:
Please follow Garden's outside block convention for CSS commenting. See https://github.com/zendeskgarden/react-components/blob/1d5b1dfe698e3692ef11a9ed5ca194cc98307274/packages/buttons/src/styled/StyledButton.ts#L47-L88 for details.